src/averageAges.js

```diff
@@ -37,7 +56,25 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const getWomen = people.filter(person => person.sex === 'f');
+
+  if (arguments.length === 1) {
+    const calculateAge2 = getWomen.map(person =>
+      person.died - person.born);
+    const calculateAvgAge2 = calculateAge2.reduce((a, b) =>
+      a + b, 0) / calculateAge2.length;
+
+    return calculateAvgAge2;
+  }
+
+  const getWomenWithChildren = getWomen.filter(person =>
+    people.some(child => child.mother === person.name));
```

Comment:
I don't understand it fully. I think some is an incorrect method to use here. Some returns true if at least one element of array pass the check. So, it won't cycle through the whole array it will return true even if only one item in the array is truthy. You are using it just to create an additional iteration over the array, but you can use a better method here, like map or find.